src/metrics.ts

```diff
@@ -63,8 +65,260 @@ function initializeMetricMaps() {
+        'bQd': {
+          initialPosition: 'd8',
+          currentPosition: 'd8',
+          distanceMoved: 0,
+        },
+        'bKe': {
+          initialPosition: 'e8',
+          currentPosition: 'e8',
+          distanceMoved: 0,
+        },
+        'bBf': {
+          initialPosition: 'f8',
+          currentPosition: 'f8',
+          distanceMoved: 0,
+        },
+        'bNg': {
+          initialPosition: 'g8',
+          currentPosition: 'g8',
+          distanceMoved: 0,
+        },
+        'bRh': {
+          initialPosition: 'h8',
+          currentPosition: 'h8',
+          distanceMoved: 0,
+        },
+        'bPa': {
+          initialPosition: 'a7',
+          currentPosition: 'a7',
+          distanceMoved: 0,
+        },
+        'bPb': {
+          initialPosition: 'b7',
+          currentPosition: 'b7',
+          distanceMoved: 0,
+        },
+        'bPc': {
+          initialPosition: 'c7',
+          currentPosition: 'c7',
+          distanceMoved: 0,
+        },
+        'bPd': {
+          initialPosition: 'd7',
+          currentPosition: 'd7',
+          distanceMoved: 0,
+        },
+        'bPe': {
+          initialPosition: 'e7',
+          currentPosition: 'e7',
+          distanceMoved: 0,
+        },
+        'bPf': {
+          initialPosition: 'f7',
+          currentPosition: 'f7',
+          distanceMoved: 0,
+        },
+        'bPg': {
+          initialPosition: 'g7',
+          currentPosition: 'g7',
+          distanceMoved: 0,
+        },
+        'bPh': {
+          initialPosition: 'h7',
+          currentPosition: 'h7',
+          distanceMoved: 0,
+        },
+      };
+
+      // Initialize variables to keep track of the maximum distance and the piece
+      let maxDistance = -1;
+      let maxDistancePiece = null;
+
+      ingestedMoves.forEach(move => {
+        
+        const fromSquare = move.from;
+        const toSquare = move.to;
+        //console.log(`fromSquare: ${fromSquare}`);
+        //console.log(`toSquare: ${toSquare}`);
+
+
+        function calculateDistance(fromSquare: string, toSquare: string) {
+          let totalDistance;
+
+          // Calculate the distance moved by the piece
+          const fileDistance = Math.abs(fromSquare.charCodeAt(0) - toSquare.charCodeAt(0));
+          const rankDistance = Math.abs(Number(fromSquare[1]) - Number(toSquare[1]));
+
+          // Check if it's a diagonal move
+          if (fileDistance === rankDistance) {
+            // This is a diagonal move, count it as a single horizontal/vertical movement
+            totalDistance = fileDistance;
+            console.log(`Total distance moved: ${totalDistance}`);
+          } 
+          else {
+            // Not a diagonal move, calculate the total distance, where one square = 1 unit of distance
+            totalDistance = fileDistance + rankDistance;
+            console.log(`Total distance moved: ${totalDistance}`);
+          }
+          return totalDistance;
```

Comment:
Also if we count diagonal as one all we actually need to do is take the greater of rank & file distance. For diagonal moves it'll be the same, for straight one will be 0 the other will be the value & for knights one is 2 & the other is 1

Reply:
true good pt